apps/server/src/modules/authorization/domain/service/authorization-reference.service.spec.ts

```diff
@@ -0,0 +1,183 @@
+					useValue: createMock<AuthorizationService>(),
+				},
+				{
+					provide: ReferenceLoader,
+					useValue: createMock<ReferenceLoader>(),
+				},
+			],
+		}).compile();
+
+		service = await module.get(AuthorizationReferenceService);
+		authorizationService = await module.get(AuthorizationService);
+		loader = await module.get(ReferenceLoader);
+	});
+
+	afterEach(() => {
+		jest.resetAllMocks();
+	});
+
+	describe('checkPermissionByReferences', () => {
+		const setupData = () => {
+			const entityId = new ObjectId().toHexString();
+			const userId = new ObjectId().toHexString();
+			const context = AuthorizationContextBuilder.read([]);
+			const entityName = AuthorizableReferenceType.Course;
+
+			return { context, entityId, userId, entityName };
+		};
+
+		describe('when hasPermissionByReferences returns false', () => {
+			const setup = () => {
+				const { entityId, userId, context, entityName } = setupData();
+
+				const spy = jest.spyOn(service, 'hasPermissionByReferences').mockResolvedValueOnce(false);
+
+				return { context, userId, entityId, entityName, spy };
+			};
+
+			it('should reject with ForbiddenLoggableException', async () => {
+				const { context, userId, entityId, entityName, spy } = setup();
+
+				await expect(service.checkPermissionByReferences(userId, entityName, entityId, context)).rejects.toThrow(
+					new ForbiddenLoggableException(userId, entityName, context)
+				);
+
+				spy.mockRestore();
+			});
+		});
+
+		describe('when hasPermissionByReferences returns true', () => {
+			const setup = () => {
+				const { entityId, userId, context, entityName } = setupData();
+
+				const spy = jest.spyOn(service, 'hasPermissionByReferences').mockResolvedValueOnce(true);
+
+				return { context, userId, entityId, entityName, spy };
+			};
+
+			it('should resolve without error', async () => {
+				const { context, userId, entityId, entityName, spy } = setup();
+
+				await expect(service.checkPermissionByReferences(userId, entityName, entityId, context)).resolves.not.toThrow();
+
+				spy.mockRestore();
+			});
+		});
+	});
+
+	describe('hasPermissionByReferences', () => {
+		const setupData = () => {
+			const entity = courseFactory.buildWithId();
+			const user = userFactory.buildWithId();
+			const context = AuthorizationContextBuilder.read([]);
+			const entityName = AuthorizableReferenceType.Course;
+
+			return { context, entity, user, entityName };
+		};
+
+		describe('when loader throws an error', () => {
+			const setup = () => {
+				const { entity, user, context, entityName } = setupData();
+
+				loader.loadAuthorizableObject.mockRejectedValueOnce(new NotFoundException());
+				authorizationService.getUserWithPermissions.mockResolvedValueOnce(user);
+				authorizationService.hasPermission.mockReturnValueOnce(true);
+
+				return { context, userId: user.id, entityId: entity.id, entityName };
+			};
+
+			it('should reject with ForbiddenLoggableException', async () => {
```

Comment:
```suggestion
			it('should reject with NotFoundException', async () => {
```

Reply:
Ich hab erstmal nur gesehen, dass die Fehler nicht zusammen passen, aber eigentlich geht es doch hier gar nicht um einen bestimmten Fehler, sondern nur darum, dass ein Fehler nicht abgefangen wird.

Um das zu verdeutlichen, würd ich eher in Zeile 107 `				loader.loadAuthorizableObject.mockRejectedValueOnce(new Error('test'));` schreiben und dann auf diesen Fehler prüfen.

Gilt auch für den Test darunter.

Reply:
Die Beschreibung habe ich angepasst. NotFoundException dagegen finde ich nicht verkeht. Es kann immer ein js error fliegen. Aber in dem genutzten Kontext ist eine NotFoundException am ehsten zu erwarten. Daher würde ich diese gerne drin lassen.

Reply:
Ich würde beim Lesen dann aber erstmal denken, dass es ne Bedeutung hat, dass das ausgerechnet ne NotFoundException ist. Während ein allgemeiner Fehler mit ner Message wie `new Error('test message')` direkt darauf hinweist, dass es nicht um irgendeinen Fehler mit ner speziellen Bedeutung geht, sondern darum, dass irgendwas mit diesem konkreten Fehler passiert. 

Der Kontext, in dem die getestete Funktion genutzt wird, sollte dem Test egal sein, würd ich meinen.